### ISSUE TITLE
Need help with ComboBox DropDown & Popup

### ISSUE THREAD

**Issue by naumenkoff** [microsoft/microsoft-ui-xaml#10060]
Hello everyone.  
I can't figure out how the ComboBox works: for some reason, it has a gray background, and I can't get rid of it.  
For example, in `NavigationView`, I able to remove it by overriding `NavigationViewContentBackground` to `Transparent`.  
But here it's a completely different situation: I've read tons of articles and looked through the code of many apps that use `ComboBox`, where it is rendered properly, but I still haven't found a solution.

**What I am referring to**:  
- In the Settings app (Windows 11), `ComboBox` has an opaque, non-gray background.  
![Image](https://github.com/user-attachments/assets/90787244-5bee-4c0c-8f7a-35d8aec2b0d2)  
- In the MVVM Toolkit Sample App (The Final Result, SubredditWidget), `ComboBox` has a transparent blurred, non-gray background.  
![Image](https://github.com/user-attachments/assets/0525e930-6ea5-4c0e-9c08-6d3789c96b14)  
- `DropDownButton`
```xml
<DropDownButton Content="Say">
    <DropDownButton.Flyout>
        <MenuFlyout Placement="Bottom">
            <MenuFlyoutItem Text="Hello" />
            <MenuFlyoutItem Text="World" />
        </MenuFlyout>
    </DropDownButton.Flyout>
</DropDownButton>
```
![Image](https://github.com/user-attachments/assets/cf6f19cb-6179-4c67-af09-d0b4e42af780)

But for me, everything looks completely different. I don't know how this happened. And why no solutions are working!

## ComboBox  
It differs from **`DropDownButton`**.
```xml
<ComboBox>
    <ComboBoxItem Content="Hello"/>
    <ComboBoxItem Content="World"/>
</ComboBox>
```
![Image](https://github.com/user-attachments/assets/2cde7ae6-7e63-455c-a922-7aeb11d2bb5a)

## ComboBox with overridden ComboBoxDropDownBackground
```xml
<ComboBox>
    <ComboBoxItem Content="Hello"/>
    <ComboBoxItem Content="World"/>
    <ComboBox.Resources>
        <SolidColorBrush x:Key="ComboBoxDropDownBackground" Color="{ThemeResource AcrylicInAppFillColorDefaultBrush}" />
    </ComboBox.Resources>
</ComboBox>
```
![Image](https://github.com/user-attachments/assets/e31c83cc-75ff-4530-b5c9-53532d91a08b)

I referred to these Issues:  
- https://github.com/microsoft/microsoft-ui-xaml/issues/7176  
- https://github.com/microsoft/microsoft-ui-xaml/issues/7200  
- https://github.com/microsoft/microsoft-ui-xaml/issues/8396  
- https://github.com/microsoft/microsoft-ui-xaml/issues/8657  
- https://github.com/microsoft/microsoft-ui-xaml/issues/9523  

## Versions:
- **Nuget: WindowsAppSDK 1.6.240923002, Runtime: 1.6.1**
- **net8.0-windows10.0.26100.0**
- **Windows 11 Pro 26100.2033**

**Comment by whiskhub** [comment]
It's a bug in WinUI 3. Currently only in WinUI 2 (UWP) there is an acrylic background, sadly.

**Comment by Lightczx** [comment]
https://github.com/DGP-Studio/Snap.Hutao/blob/main/src/Snap.Hutao/Snap.Hutao/UI/Xaml/Behavior/ComboBoxSystemBackdropWorkaroundBehavior.cs
Just use `ContentExternalBackdropLink`

**Comment by hexawyz** [comment]
For what it's worth, I have used the code from @Lightczx and it does work, however, **NOT** if the app is AOT-compiled. (Then nothing happens, as if the code wasn't there)
I tried investigating a bit and I assume some of the internals of WinUI are different (trimmed ?) which makes the code unable to do what it tries to do. (I couldn't confirm with enough confidence due to the limited debugging experience available in AOT, but it seemed like the code might never find the `Popup` in that case)
The workaround is also somewhat slow in some situations (e.g. switching to a (cached?) page with a dozen `ComboBox` will freeze for a noticeable amount of time). I assume that this is related to some of the component allocations and swapping around of the UI tree that takes place in the workaround. As none of that would be needed if this was directly supported by WinUI, I assume it would be at least somewhat faster in that case.
Additionally, It would seem that the workaround entirely removes the background texture instead of mixing it like it is supposed to be (In the W11 settings app, if you go to the Date & Time page, you can easily see both texture and background. Altough the texture used there is possibly more subtle than the aggressive one provided by WinUI 3)
This may not be a problem with the workaround itself, though, as WinUI3 doens't seem to ever be able to do this mixing between texture and background anywhere except in the mostly useless "in-app" `AcrylicBrush`, as actionable solutions for other flyouts also require setting the background to transparent (which comes with other visual problems…), thus removing the texture.

Honestly, I would never have spent the tiniest amount of time trying to understand WinRT internals and make this work, so I'm just glad that someone took time to try understanding these cryptic APIs and provide a workaround in a working state (except for AOT). Thanks @Lightczx !
Hopefully, this will help getting a proper native WinUI 3 implementation of that feature that has been missing for years now, which would benefit everybody.

**Comment by Lightczx** [comment]
Oh, The code I use is credit to @HO-COOH
I just somehow translate his code from C++ to C#

**Comment by Lightczx** [comment]
> In the W11 settings app, if you go to the Date & Time page, you can easily see both texture and background.

The settings app uses CoreWindow, and the popup create by combobox is constrained to root bound by default. And so it uses in-app acrylic brush.
But for WASDK WinUI 3, the combobox creates the popup that doesn't constrained to root bound by default(set in style).
And I believe it's intentional done because they want to avoid the issue that combobox's popup can be overlapped(covered) by titlebar if you set `ExtendsContentIntoTitleBar` to true and that makes some items unclickable.

**Comment by Lightczx** [comment]
> Everything returned is typed explicitly as `FrameworkElement` instead of the actual derived types.

Yeah, use TryAs<Popup> to check the actual type since the native abi should stay the same between AOT and not.
Actually you can implment TryAs like this (I'm not sure whether a using wrap for the objRef is needed):
```
((IWinRTObject)element).NativeObject.TryAs(out ObjectReference<T> objRef);
xxx = MarshalInspectable<T>.FromAbi(objRef.ThisPtr);
```
